grammar.js

```diff
@@ -193,35 +193,40 @@ module.exports = grammar({
     assert_msg: ($) => seq(",", $.expr),
 
     type: ($) =>
+      choice(
+        $.simple_type,
+        "timer",
+        seq("union", "{", list1($.type, ","), "}"),
+        seq("function", $.func_params),
+        seq("event", "(", optional($.formal_args), ")"),
+        seq("hook", "(", optional($.formal_args), ")"),
+        $.id,
+      ),
+
+    simple_type: ($) =>
```

Comment:
This rule is more an implementation detail and not something interesting for downstream consumers, e.g., I couldn't imagine what special thing we would do in https://github.com/zeek/zeekscript for `simple_type` as opposed to just `type`.

I'd suggest to instead use a hidden node, i.e., `_simple_type` as opposed to `simple_type` here and everywhere else.